assets/src/components/form/select/build-options.js

```diff
@@ -1,65 +1,59 @@
 /**
  * External dependencies
  */
-import { reduce, isFunction } from 'lodash';
-
-/**
- * Internal dependencies
- */
-import { dateTimeModel } from '../../../data/model';
-
-/**
- * A default map used for mapping options for select.
- * @type {Object}
- */
-const DEFAULT_MODEL_OPTIONS_MAP = {
-	event: {
-		label: 'EVT_name',
-		value: 'EVT_ID',
-	},
-	datetime: {
-		label: ( entity ) => {
-			return dateTimeModel.prettyDateFromDateTime( entity );
-		},
-		value: 'DTT_ID',
-	},
-	ticket: {
-		label: 'TKT_name',
-		value: 'TKT_ID',
-	},
-};
+import { isEmpty, isFunction, isUndefined, reduce } from 'lodash';
 
 /**
  * Receives an array of event entities and returns an array of simple objects
  * that can be passed along to the options array used for the WordPress
  * SelectControl component.
  *
- * @param { Array } entities
- * @param { string } modelName
- * @param { Object } map
- * @return { Array }  Returns an array of simple objects formatted for any
- * select control that receives its options in the format of an array of objects
- * with label and value keys.
+ * @param { Array }  entities
+ * @param { Object } optionsEntityMap 	Object of key value pairs where values are the
+ * 										'label:value' pairings used to populate the Select input
+ * @param { string } mapSelection		Determines which optionEntityMap pairing to use
+ * @return { Array }  					Returns an array of simple objects formatted for any
+ * 										select control that receives its options
+ * 										in the format of an array of objects
+ *										with label and value keys.
  */
 const buildOptions = (
 	entities,
-	modelName,
-	map = DEFAULT_MODEL_OPTIONS_MAP,
+	optionsEntityMap,
+	mapSelection = 'default'
 ) => {
-	const MAP_FOR_MODEL = map[ modelName ] ? map[ modelName ] : false;
-	return entities && MAP_FOR_MODEL ?
-		reduce( entities, function( options, entity ) {
-			const label = isFunction( MAP_FOR_MODEL.label ) ?
-				MAP_FOR_MODEL.label( entity ) :
-				entity[ MAP_FOR_MODEL.label ];
-			const value = isFunction( MAP_FOR_MODEL.value ) ?
-				MAP_FOR_MODEL.value( entity ) :
-				entity[ MAP_FOR_MODEL.value ];
-			if ( label && value ) {
-				options.push( { label, value } );
-			}
-			return options;
-		}, [] ) :
+	if ( isEmpty( entities ) || isEmpty( optionsEntityMap ) ) {
+		return [];
+	}
+	// if requested mapSelection exists then use that
+	let map = ! isUndefined( optionsEntityMap[ mapSelection ] ) ?
+		optionsEntityMap[ mapSelection ] :
+		null;
+	// if not but default exists then use that
+	map = map === null && ! isUndefined( optionsEntityMap.default ) ?
+		optionsEntityMap.default :
+		null;
+	// if default doesn't exist then just grab first option
+	map = map === null ?
+		optionsEntityMap[ Object.keys( optionsEntityMap )[ 0 ] ] :
+		map;
```

Comment:
I don't think we should use a fallback to `default` (or the first option) if the requested `mapSelection` value doesn't have a match.  That could lead to hidden unexpected and hard to trace behaviour.  Instead I think it'd be better to maybe print a console warning here (using the `warn` module`) and just return an empty array (btw you can also unit-test for console behaviour... if you need help with that let me know.

If you make my suggested changes, this would effectively mean that EVERY provided map must be at _least_ indexed by a `default` key.  Since that is the default value for the `mapSelection` argument, that allows for correctly reducing to an option array if an optionMap is provided with no `mapSelection` argument.  At the same time, there will be helpful warnings given in the console (in development mode) if an invalid `optionMap` or `mapSelection` argument is provided.